bilby/gw/likelihood.py

```diff
@@ -1526,76 +1529,53 @@ def log_likelihood_ratio(self):
         d_inner_h = 0.
         optimal_snr_squared = 0.
         complex_matched_filter_snr = 0.
+
         self.parameters.update(self.get_sky_frame_parameters())
+
         waveform_ratio = self.compute_waveform_ratio(self.parameters)
 
         for interferometer in self.interferometers:
-            per_detector_snr = self.calculate_snrs_relative_binning(waveform_ratio[interferometer.name], interferometer)
-
+            per_detector_snr = self.calculate_snrs_relative_binning(
+                waveform_ratio[interferometer.name], interferometer)
             d_inner_h += per_detector_snr.d_inner_h
-            optimal_snr_squared += np.real(
-                per_detector_snr.optimal_snr_squared)
+            optimal_snr_squared += np.real(per_detector_snr.optimal_snr_squared)
             complex_matched_filter_snr += per_detector_snr.complex_matched_filter_snr
 
         log_l = np.real(d_inner_h) - optimal_snr_squared / 2
         return float(log_l.real)
 
-    def find_maximum_likelihood_waveform(self, initial_parameter_guess,
-                                         parameter_bounds, iterations=10,
-                                         likelihood_threshold=1):
-        prev_log_likelihood = -np.inf
+    def find_maximum_likelihood_parameters(self, parameter_bounds,
+                                           iterations=1, atol=1e-3, maxiter=100):
+
+        parameter_bounds_list = self.get_parameter_list_from_dictionary(parameter_bounds)
+
         for i in range(iterations):
-            print("iter: %s" % i)
-            print("length of parameter_bounds", len(parameter_bounds))
-            log_likelihood = self.get_best_fit_parameters(
-                self.get_parameter_list_from_dictionary(parameter_bounds),
-                atol=1e-10, maxiter=10)  # change back to 500, no input
-            print("likelihood: %s" % log_likelihood)
-
-            self.maximum_likelihood_parameters.update(self.get_sky_frame_parameters())
-            self.set_fiducial_waveforms(self.maximum_likelihood_parameters)
+            logger.info("Optimizing fiducial parameters. Iteration : {}".format(i + 1))
+            output = differential_evolution(self.lnlike_scipy_maximize,
+                                            bounds=parameter_bounds_list, atol=atol, maxiter=maxiter)
```

Comment:
So I think one question we need to answer is: how many waveform evaluations is `differential_evolution` doing per iteration. If the run takes a total of 15 minutes, then that would mean ~30000 likelihood evaluations in `differential_evolution` (assuming each evaluation takes 3e-2 seconds).

Do you know the answer to this?

Reply:
I guess we do know the answe to this now :smile: